sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusErrorSource.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.ExpandableStringEnum;
+
+import java.io.Serializable;
+
+/**
+ * Represent the operation user was performing when the error happened.
+ */
+public final class ServiceBusErrorSource extends ExpandableStringEnum<ServiceBusErrorSource> implements Serializable {
+
+    private static final long serialVersionUID = -2819764417333954922L;
+
+    /** Error while abandoning the message.*/
+    public static final ServiceBusErrorSource ABANDONED = fromString("ABANDONED", ServiceBusErrorSource.class);
+
+    /** Error while completing the message.*/
+    public static final ServiceBusErrorSource COMPLETE = fromString("COMPLETE", ServiceBusErrorSource.class);
+
+    /** Error while deferring the message.*/
+    public static final ServiceBusErrorSource DEFER = fromString("DEFER", ServiceBusErrorSource.class);
+
+    /** Error while dead-lettering the message.*/
+    public static final ServiceBusErrorSource DEAD_LETTER = fromString("DEAD_LETTER",
+        ServiceBusErrorSource.class);
+
+    /** Error while receiving the message(s).*/
+    public static final ServiceBusErrorSource RECEIVE = fromString("RECEIVE", ServiceBusErrorSource.class);
+
+    /** Error while renewing lock.*/
+    public static final ServiceBusErrorSource RENEW_LOCK = fromString("RENEW_LOCK", ServiceBusErrorSource.class);
```

Comment:
We'll need error sources for session-related operations too.

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/93739d8ef58c6a2d3d5263a043461a36481b270d/sdk/servicebus/Azure.Messaging.ServiceBus/api/Azure.Messaging.ServiceBus.netstandard2.0.cs#L109   Is this what  you want accept, close session ?